syft/pkg/cataloger/java/archive_filename.go

```diff
@@ -113,7 +113,7 @@ func (a archiveFilename) extension() string {
 
 func (a archiveFilename) pkgType() pkg.Type {
 	switch strings.ToLower(a.extension()) {
-	case "jar", "war", "ear", "lpkg":
+	case "jar", "war", "ear", "lpkg", "par":
```

Comment:
I'm not to familiar with the `parchive` format, but I did some initial research and found [this](https://github.com/Parchive/par2cmdline) as a starting point. 

Should we be matching on `par2` as well?
Is it correct to always associate `.par` extensions with `pkg.JavaPkg`?

I don't know the answers to these just yet, but wanted to surface these questions as part of this PR discussion.

Reply:
I'm definitely not an expert on this either, I was mostly going based off of the examples from other scanners, but I expect this isn't parchive but rather the Sun Portal Server format at https://docs.oracle.com/cd/E19528-01/819-5030/gbjsz/index.html, which is just another format containing .jar files

Reply:
Also, this isn't entirely related, but I did have a thought that it might be useful in the future to extend the catalogers to allow matching on custom patterns via config

Reply:
And searching a bit more it seems like .par might be used frequently with SAP applications

Reply:
Ahh that makes much more sense. Thanks for more context here. I'll see if I can assemble one as well and try and scan it with the changes. You're right that finding real examples of them has proven difficult.